src/main/java/org/eclipsefoundation/marketplace/service/CachingService.java

```diff
@@ -0,0 +1,26 @@
+/* Copyright (c) 2019 Eclipse Foundation and others.
+ * This program and the accompanying materials are made available
+ * under the terms of the Eclipse Public License 2.0
+ * which is available at http://www.eclipse.org/legal/epl-v20.html,
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipsefoundation.marketplace.service;
+
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Interface defining the caching service to be used within the application.
+ * 
+ * @author Martin Lowe
+ */
+public interface CachingService {
+
+	Optional<Object> get(String key);
+
+	<T> void put(String key, T value);
+	
+	void expire(String key);
```

Comment:
what's the usage of this method? If it's only for testing, make it an implementation package-private method, not one on the interface

Reply:
Ideally we'd invalidate entries when we insert new data via PUT calls, so we can attempt to keep somewhat accurate data even just after an update. The functionality just isn't fully done yet. This was more of a first pass in getting basic caching working

Reply:
gotcha. 

My moto about _preemptive_ dev: do not do stuff that you may need later, usually you will need something else.